internal/compass/compass.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compass
+
+import (
+	"errors"
+	"os/exec"
+	"time"
+)
+
+const waitForRunningStateSeconds = 10
+
+func binPath() string {
+	if p, err := exec.LookPath(compassBin); err == nil {
+		return p
+	}
+	return ""
+}
+
+func Detect() bool {
+	return binPath() != ""
+}
+
+func Run(username, password, mongoURI string) error {
+	args := []string{mongoURI}
+	if username != "" && password != "" {
+		args = append(args, "--username", username, "--password", password)
+	}
+
+	path := binPath()
+	if path != compassBin {
+		path += compassBin
+	}
+
+	cmd := exec.Command(path, args...)
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	timer := time.NewTimer(waitForRunningStateSeconds * time.Second)
```

Comment:
why do we need this?

can't we start and never wait?

Reply:
found this while testing in a ubuntu VM (ssh connection):
`exec.Command(path, args...).Start()` starts the process (so it returns immediately with no err) but Compass can fail immediately, in this case with `Missing X server or $DISPLAY`
by waiting for a few seconds and check it's status, we can see at least if the minimum requirements to run Compass are met.

Reply:
can we use context as opposed to timer? also we are waiting 10s, are we always waiting 10s before closing? or is compass forking?

Reply:
https://pkg.go.dev/context#WithTimeout

Reply:
changed to the hybrid ctx + one channel